exercises/practice/robot-simulator/.meta/example.awk

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env gawk -f
+
+BEGIN {
+    OK = 0
+    FAIL = 1
+    status = OK
+
+    # Input variables: x, y, dir
+    # handle default values
+    x = 0 + x
+    y = 0 + y
+    if (dir == "") dir = "north"
+
+    if (!(dir == "north" || dir == "east" || dir == "south" || dir == "west")) {
```

Comment:
Would this be more or less readable?

`if (dir !~ /north|east|south|west/)`

Reply:
I see what you mean. I tend to use equality for discrete items. I'd want to use anchors which reduces the readability a touch
```awk
if (dir !~ /^(north|east|south|west)$/)
```
I think I'll leave it as it: I treat the example solution as just proof that it can be solved, not necessarily an _exemplary_ solution.

I originally had this, which I thought was slightly overkill
```awk
dirs["north"] = 1
dirs["east"] = 1
dirs["south"] = 1
dirs["west"] = 1
if (!(dir in dirs)) ...
```